src/Categories/Adjoint/Instance/Slice.agda

```diff
@@ -44,3 +52,94 @@ module _ {A : Obj} (product : ∀ {X} → Product A X) where
       ⟨ π₁ , π₂ ⟩                               ≈⟨ η ⟩
       id                                        ∎
     }
+
+module _ {A : Obj} (ccc : CartesianClosed C) (pullback : ∀ {X} {Y} {Z} (h : X ⇒ Z) (i : Y ⇒ Z) → Pullback h i) where
+
+  open CartesianClosed ccc
+  open Cartesian cartesian
+  open Terminal terminal
+  open BinaryProducts products
+
+  Free⊣Coforgetful : Free {A = A} product ⊣ Coforgetful ccc pullback
+  Free⊣Coforgetful = record
+    { unit = ntHelper record
+      { η = λ X → p.universal (sliceobj π₁) (λ-unique₂′ (unit-pb X))
+      ; commute = λ {S} {T} f → p.unique-diagram (sliceobj π₁) !-unique₂ $ begin
```

Comment:
For these longer proofs, best to put them either in `where` or named in a private block. Purely for efficiency.